guides/keras_cv/retina_net_overview.py

```diff
@@ -0,0 +1,313 @@
+"""
+Title: Train an Object Detection Model on Pascal VOC 2007 using KerasCV
+Author: [lukewood](https://lukewood.xyz)
+Date created: 2022/08/22
+Last modified: 2022/08/22
+Description: Use KerasCV to train a RetinaNet on Pascal VOC 2007.
+"""
+
+"""
+## Overview
+
+KerasCV offers a complete set of APIs to train your own state-of-the-art,
+production-grade object detection model.  These APIs include object detection specific
+data augmentation techniques, models, and COCO metrics.
+
+To get started, let's sort out all of our imports and define global configuration parameters.
+"""
+
+import matplotlib.pyplot as plt
+import tensorflow as tf
+import tensorflow_datasets as tfds
+from tensorflow import keras
+from tensorflow.keras import optimizers
+
+import keras_cv
+from keras_cv import bounding_box
+
+BATCH_SIZE = 9
+EPOCHS = 100
+
+"""
+## Data loading
+
+In this guide, we use the data-loading function: `keras_cv.datasets.pascal_voc.load()`.
+KerasCV supports a `bounding_box_format` argument in all components that process
+bounding boxes.  To match the KerasCV API style, it is recommended that when writing a
+custom data loader, you also support a `bounding_box_format` argument.
+This makes it clear to those invoking your data loader what format the bounding boxes
+are in.
+
+For example:
+
+```python
+train_ds, ds_info = keras_cv.datasets.pascal_voc.load(split='train', bounding_box_format='xywh', batch_size=8)
+```
+
+Clearly yields bounding boxes in the format `xywh`.  You can read more about
+KerasCV bounding box formats [in the API docs](https://keras.io/api/keras_cv/bounding_box/formats/).
+
+Our data comesloaded into the format
+`{"images": images, "bounding_boxes": bounding_boxes}`.  This format is supported in all
+KerasCV preprocessing components.
+
+Let's load some data and verify that our data looks as we expect it to.
+"""
+
+dataset, dataset_info = keras_cv.datasets.pascal_voc.load(
+    split="train", bounding_box_format="xywh", batch_size=9
```

Comment:
Why 9 rather than 8? A priori 8 will lead to better GPU memory management

Reply:
Ah I see, I was trying to support the visualization at the end which requires 9 samples